### PR TITLE
feat: bb8-based SolanaRpcPool

### PR DESCRIPTION
**Pull request by sergeytimoshin**
Replace the old RpcPool implementation with bb8-based connection pooling for improved resource management.

### CODE REVIEW

File: forester/src/lib.rs [hunk 1]
```diff
@@ -62,26 +62,33 @@ pub async fn run_pipeline<R: RpcConnection, I: Indexer<R>>(
     shutdown: oneshot::Receiver<()>,
     work_report_sender: mpsc::Sender<WorkReport>,
 ) -> Result<(), ForesterError> {
-    let rpc_pool = Arc::new(RpcPool::<R>::new(config.clone()).await);
+    // let rpc_pool_size = Arc::new(RpcPool::<R>::new(config.clone()).await);
```
Comment (vadorovsky) [inline forester/src/lib.rs:65]: can this be removed?

**Review by vadorovsky — APPROVED** [review]
Just one nit.

File: forester/tests/e2e_test.rs [hunk 2]
```diff
@@ -335,17 +352,18 @@ async fn test_epoch_monitor_with_2_foresters() {
     assert!(total_processed > 0, "No items were processed");
 
     for tree in state_trees {
-        let queue_length =
-            fetch_queue_item_data(pool.get_connection().await, &tree.nullifier_queue)
-                .await
-                .unwrap()
-                .len();
+        let mut rpc = pool.get_connection().await.unwrap();
```
Comment (ananas-block) [inline forester/tests/e2e_test.rs:355]: can we map such errors instead of unwrapping, so that we can restart in case all connections drop for some reason?

Comment (ananas-block) [reply forester/tests/e2e_test.rs:355]: (There are a lot of unwraps in other places not commenting those explicitly.)

**Review by ananas-block — COMMENTED** [review]
Thanks great improvement looks good overall!
Only nit is that it would be great to weed out unwraps from the start. :)

File: forester/tests/e2e_test.rs (hunk 2 above)
Comment (sergeytimoshin) [reply forester/tests/e2e_test.rs:355]: I think it's safe enough. 
All the unwraps like this are only in the e2e test, forester maps [errors like this](https://github.com/Lightprotocol/light-protocol/blob/sergey/bb8-solana-rpc-pool/forester/src/epoch_manager.rs#L191).
And we also have a helper method for the case when all connections are busy: [get_connection_with_retry](https://github.com/Lightprotocol/light-protocol/blob/sergey/bb8-solana-rpc-pool/forester/src/rpc_pool.rs#L85)

Comment (ananas-block) [reply forester/tests/e2e_test.rs:355]: you are right unwraps are just in tests. We have a lot of unwraps in other places though (not related to this pr), but we can remove those in a different pr.

**Pull request merged**